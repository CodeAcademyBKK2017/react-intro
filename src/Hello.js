import React from 'react';
class Hello extends React.Component{
    render(){
        const cel = {
            margin:'1px',
            backgroundColor: '#2c98f0',
            display: 'inline-block',
            padding: '15px 10px',
            borderradius: '3px',
            color:'#fff',
            cursor: 'pointer',
        }
        const row = {
            
        }
        return(
            <div>
                <div style={row}>
                    <div style={cel}>
                        {this.props.name}
                    </div>
                    <div style={cel}>
                        {this.props.name}
                    </div>
                    <div style={cel}>
                        {this.props.name}
                    </div>
                </div>
                <div style={row}>
                    <div style={cel}>
                        {this.props.name}
                    </div>
                    <div style={cel}>
                        {this.props.name}
                    </div>
                    <div style={cel}>
                        {this.props.name}
                    </div>
                </div>
                <div style={row}>
                    <div style={cel}>
                        {this.props.name}
                    </div>
                    <div style={cel}>
                        {this.props.name}
                    </div>
                    <div style={cel}>
                        {this.props.name}
                    </div>
                </div>
            </div>
        );
    }
}
export default Hello;