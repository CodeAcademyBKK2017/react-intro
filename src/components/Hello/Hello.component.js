import React, { Component } from 'react';
import PropTypes from 'prop-types';

class Hello extends Component {
	render() {

		// console.log(this.props);

		// return (
		//   <div>
		//     <div>
		//       Hello
		//     </div>
		//     <div>
		//       {this.props.name} World
		//     </div>
		//   </div>
		// );

		const hello = <div key="1">Hello</div>
		const world = <div key="2" >{this.props.name} World</div>
		return [hello, world];
	}
}

Hello.propTypes = {
	name: PropTypes.string.isRequired
};

export default Hello;
