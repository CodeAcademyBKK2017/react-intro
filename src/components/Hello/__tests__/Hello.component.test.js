import React from 'react';
import Hello from '../Hello.component';
import renderer from 'react-test-renderer';

test('Hello', () => {
	const data = 'chong';
	const snapshot = renderer.create(<Hello name={data} />).toJSON();
	expect(snapshot).toMatchSnapshot();
});
