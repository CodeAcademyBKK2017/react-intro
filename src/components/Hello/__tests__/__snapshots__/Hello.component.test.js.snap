// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hello 1`] = `
Array [
  <div>
    Hello
  </div>,
  <div>
    chong
     World
  </div>,
]
`;
